k8s/scripts/inspect.sh

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env bash
+
+INSPECT_DUMP=$(pwd)/inspection-report
+
+SVC_ARGS_DIR=/var/snap/k8s/common/args
+K8SD_STATE_DIR=/var/snap/k8s/common/var/lib/k8sd/state
+K8SD_BIN=/snap/k8s/current/bin/k8sd
+SBOM_FILE=/snap/k8s/current/bom.json
+
+function log_success {
+    printf -- '\033[32m SUCCESS: \033[0m %s\n' "$1"
+}
+
+function log_info {
+	printf -- '\033[34m INFO: \033[0m %s\n' "$1"
+}
+
+function log_warning_red {
+	printf -- '\033[31m WARNING: \033[0m %s\n' "$1"
+}
+
+function collect_args {
+	log_info "Copy service args to the final report tarball"
+	cp -r --no-preserve=mode,ownership "$SVC_ARGS_DIR" "$INSPECT_DUMP"
+}
+
+function collect_cluster_info {
+	log_info "Copy k8s cluster-info dump to the final report tarball"
+	k8s kubectl cluster-info dump &> "$INSPECT_DUMP"/cluster-info
+}
+
+function collect_sbom {
+	log_info "Copy SBOM to the final report tarball"
+	cp --no-preserve=mode,ownership "$SBOM_FILE" "$INSPECT_DUMP"/sbom.json
+}
+
+function collect_diagnostics {
+	log_info "Copy uname to the final report tarball"
+	uname -a &> "$INSPECT_DUMP"/uname
+
+	log_info "Copy snap diagnostics to the final report tarball"
+	snap version &> "$INSPECT_DUMP"/snap-version
+	snap list k8s &> "$INSPECT_DUMP"/snap-list-k8s
+	snap services k8s &> "$INSPECT_DUMP"/snap-services-k8s
+	snap logs k8s -n 10000 &> "$INSPECT_DUMP"/snap-logs-k8s
+
+	log_info "Copy k8s diagnostics to the final report tarball"
+	k8s version &> "$INSPECT_DUMP"/k8s-version
+	k8s status &> "$INSPECT_DUMP"/k8s-status
+	k8s get &> "$INSPECT_DUMP"/k8s-get
+	k8s kubectl get cm k8sd-config -n kube-system -o yaml &> "$INSPECT_DUMP"/k8sd-configmap
+	k8s kubectl get cm -n kube-system &> "$INSPECT_DUMP"/k8s-configmaps
+
+	cp --no-preserve=mode,ownership /var/snap/k8s/common/var/lib/k8s-dqlite/cluster.yaml "$INSPECT_DUMP"/k8s-dqlite-cluster.yaml
+	cp --no-preserve=mode,ownership /var/snap/k8s/common/var/lib/k8sd/state/database/cluster.yaml "$INSPECT_DUMP"/k8sd-cluster.yaml
+}
+
+function check_service {
+	local service=$1
+	mkdir -p "$INSPECT_DUMP"/"$service"
+
+	local status_file="$INSPECT_DUMP/$service/systemctl.log"
+
+    systemctl status "snap.$service" &> "$status_file"	
+	
+    if grep -q "active (running)" "$status_file"; then
+        log_info "Service $service is running"
+    else
+        log_info "Service $service is not running"
+    fi
+
+	journalctl -n 100000 -u "snap.$service" &> "$INSPECT_DUMP/$service/journal.log"
```

Comment:
we also need some way to note if services are _expected_ to run.

e.g. if we are on a control plane node, the apiserver-proxy service is not expected to run.

further, add a check for the `NRestart` property, which is non-zero if the service is constantly restarting due to errors. That's a useful metric to have